chain-impl-mockchain/src/leadership/genesis/mod.rs

```diff
@@ -662,4 +663,34 @@ mod tests {
 
         assert!(selection.verify(&block.header).failure());
     }
+
+    #[test]
+    pub fn leadership_not_in_the_current_epoch() {
+        let date = BlockDate {
+            epoch: 2,
+            slot_id: 0,
+        };
+        let ledger = LedgerBuilder::from_config(ConfigBuilder::new(0))
+            .build()
+            .expect("cannot build test ledger")
+            .ledger;
+
+        let stake_pool = StakePoolBuilder::new().build();
+        ledger
+            .delegation()
+            .register_stake_pool(stake_pool.info())
+            .expect("cannot register stake pool");
+        let selection = LeadershipData::new(date.epoch, &ledger);
+        let rng = rand_core::OsRng;
+        let sk = &SecretKey::generate(rng);
+        let header = HeaderBuilderNew::new(BlockVersion::Ed25519Signed, &Contents::empty())
+            .set_parent(&HeaderId::zero_hash(), ledger.chain_length().next())
```

Comment:
`.increase()` instead of `.next()` so that `property::ChainLength` is not needed.